indexd/src/app_client.rs

```diff
@@ -317,14 +346,79 @@ impl Client {
         state.finalize().into()
     }
 
+    /// Creates a signed url that can be shared with others
+    /// to give read access to a single object. An expired
+    /// link does not necessarily remove access to an object.
+    ///
+    /// # Arguments
+    /// - `object_key` the key of the object
+    /// - `encryption_key` the key used to encrypt the metadata
+    /// - `valid_until` the time the link expires
+    pub fn object_share_url(
+        &self,
+        object_key: &Hash256,
+        encryption_key: [u8; 32],
+        valid_until: OffsetDateTime,
+    ) -> Result<String> {
```

Comment:
I'd return a `Url` here. That makes the rust SDK slightly nicer and in the FFI we can use a string.